operator-common/src/test/java/io/strimzi/operator/PlatformFeaturesAvailabilityTest.java

```diff
@@ -7,157 +7,155 @@
 import io.fabric8.kubernetes.client.DefaultKubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.VersionInfo;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
 import io.vertx.core.Vertx;
 import io.vertx.core.http.HttpMethod;
 import io.vertx.core.http.HttpServer;
-import io.vertx.junit5.Checkpoint;
 import io.vertx.junit5.VertxExtension;
 import io.vertx.junit5.VertxTestContext;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 
 import java.text.ParseException;
-import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
 
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
 
 
 @ExtendWith(VertxExtension.class)
 public class PlatformFeaturesAvailabilityTest {
-    protected static Vertx vertx;
 
-    @BeforeAll
-    public static void before() {
-        vertx = Vertx.vertx();
+    private Supplier<List<String>> apis;
+    private Supplier<String> version;
+
+    private HttpServer mockHttp;
+
+    private final String defaultVersion =   "{\n" +
+                                            "  \"major\": \"1\",\n" +
+                                            "  \"minor\": \"16\",\n" +
+                                            "  \"gitVersion\": \"v1.9.1+a0ce1bc657\",\n" +
+                                            "  \"gitCommit\": \"a0ce1bc\",\n" +
+                                            "  \"gitTreeState\": \"clean\",\n" +
+                                            "  \"buildDate\": \"2018-06-24T01:54:00Z\",\n" +
+                                            "  \"goVersion\": \"go1.9\",\n" +
+                                            "  \"compiler\": \"gc\",\n" +
+                                            "  \"platform\": \"linux/amd64\"\n" +
+                                            "}";
+    private Vertx vertx;
+    private KubernetesClient kubernetesClient;
+
+    @BeforeEach
+    void setup(Vertx vertx) throws ExecutionException, InterruptedException {
+        this.vertx = vertx;
+
+        //Default to no apis.
+        apis = List::of;
+        version = () -> defaultVersion;
```

Comment:
What does this do?

Reply:
Overnight I realized that you were using suppliers or something like that here. The code would be much more readable if you call it `versionSupplier` and not just `version`.